examples/flutter_gallery/lib/demo/animation/home.dart

```diff
@@ -0,0 +1,533 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Based on https://material.uplabs.com/posts/google-newsstand-navigation-pattern
+// See also: https://material-motion.github.io/material-motion/documentation/
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+import 'sections.dart';
+import 'widgets.dart';
+
+const Color _kAppBackgroundColor = const Color(0xFF353662);
+
+// This app's contents start out at _kHeadingMaxHeight and they function like
+// an appbar. Initially the appbar occupies most of the screen and its section
+// headings are laid out in a column. By the time its height has been
+// reduced to _kAppBarMidHeight, its layout is horizontal, only one section
+// heading is visible, and the section's list of details is visible below the
+// heading. The appbar's height can be reduced to no more than _kAppBarMinHeight.
+final double _kAppBarMinHeight = 90.0;
+final double _kAppBarMidHeight = 256.0;
+// The AppBar's max height depends on the screen, see AnimationDemoHomeState._buildBody()
+
+// Initially occupies the same space as the status bar and gets smaller as
+// the primary scrollable scrolls upwards.
+class _RenderStatusBarPaddingSliver extends RenderSliver {
```

Comment:
This seems like a generally useful sliver that we'll probably want in the framework.  Maybe add a TODO about moving it into the framework?